### DISCUSSION TITLE
Nuxt instance not available on calling useFetch twice inside a composable

### DISCUSSION THREAD

**Discussion by awais019** [nuxt/nuxt#20810] [Questions] [answered]
I'm trying to catch a 401 error and then call the refresh token API to obtain a new access token before processing the original request. However, I encountered an error stating that the Nuxt instance is not available on the server side. If you guys have any other workaround for this use case then let me know.
```typescript
import { useUserStore } from "~~/stores/userStore";
import { API_Response } from "~~/types/api";

export const useFetchWrapper = async <T>(url: string, options: any) => {
  const userStore = useUserStore();
  const router = useRouter();
  const baseURL = useRuntimeConfig().public.baseURL;
  const response = await useFetch<T>(url, {
    ...options,
    headers: {
      Authorization: useUserStore().authToken,
    },
    baseURL,
  });


  if (response.error.value?.statusCode == 401) {
    const { data, error } = await useFetch<
      API_Response<{ token: string; refreshToken: string }>
    >("/auth/resfresh/token", {
      method: "POST",
      body: {
        refreshToken: userStore.refreshToken,
      },
      baseURL: useRuntimeConfig().public.baseURL,
    });
    if (data.value) {
      userStore.setAuthandRefreshToken(
        data.value.data.token,
        data.value.data.refreshToken
      );
    }
  }

  return response;
};

```

**Comment by awais019** [comment]
![Screenshot from 2023-05-12 16-49-02](https://github.com/nuxt/nuxt/assets/75579534/6d4eee59-e82f-4494-ad6c-d3ca9d70648d)

**Comment by AloisSeckar** [comment]
I believe you are running into the same situation as is [described and explained here](https://github.com/nuxt/nuxt/issues/14269#issuecomment-1397352832). 

The post also contains a possible solution using `callWithNuxt` utility, although I am not exacty sure how to interpolate it for your exact case. I think you may try to wrap your token-refresh logic into a function and supply `callWithNuxt` with it.

**Comment by warflash** [comment, marked as answer]
Where us this being called from? Also, you might want to give $fetch a shot instead of useFetch

**Comment by system012** [comment]
I believe the problem lies with the use of `useRuntimeConfig().public.baseURL` within the second useFetch.

Since you already saved the result to the variable `baseURL` , you can try using said variable instead of calling `useRuntimeConfig` a second time and it should work.
